First pass at Aave deposit

priyankaranke opened the pull request:
contracts/interfaces/ contains all the interfaces I believe we will need. (We don't need the entire interfaces but I've kept it in for now so I don't have to re-add function signatures I might need later)

contracts/FanDonation.sol contains the smart contract and currently takes USDC and deposits it into an Aave Lending Pool. 

Next step is to test this on Kovan, followed by write actual tests under test/. After this, I will work on 1inch integration/smart contract for 'Cash Out' on the creator page (if needed).

janclarin committed:
Add support creator form to fan page

janclarin committed:
Add fan support card

Supports two states: connect wallet and send. Automatically populates
address from URL query params.

janclarin committed:
Update html title and description

janclarin committed:
Add connect wallet button handler prop to Fan

janclarin committed:
Add @testing-library/react to dev dependencies

janclarin committed:
Fix connect/send button logic

janclarin committed:
Add tests for Fan

janclarin committed:
Use toBeDisabled/Enabled expect functions

janclarin committed:
Add solhint to dev dependencies

janclarin committed:
Add npm-run-all lint and fix scripts

janclarin committed:
Add solhint config files

chailey committed:
[feature] Changed the examples page and added a wallet connect function

chailey committed:
[feature] Changed the examples page and added a wallet connect function

chailey committed:
[feature] Changed the examples page and added a wallet connect function

chailey committed:
[feature] Changed the examples page and added a wallet connect function

janclarin committed:
Apply solhint fixes

janclarin committed:
Update truffle solc version to 0.6.12

priyankaranke committed:
First pass at Aave deposit

priyankaranke committed:
First pass at Aave deposit

priyankaranke committed:
Add missing files to not break build

priyankaranke committed:
Add missing files to not break build

priyankaranke committed:
Fixed file path

priyankaranke committed:
Fixed file path

janclarin commented on contracts/FanDonation.sol:
```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: agpl-3.0
```
nit: We're using MIT for the repo, so let's substitute agpl-3.0 with MIT for all of our authored contracts.

janclarin commented on contracts/FanDonation.sol:
```diff
@@ -0,0 +1,30 @@
+
+    // Set the address of the AAVE protocol addresses provider. Should not change once deployed.
+    // Currently set to the LendingPoolAddressesProvider for Kovan (https://docs.aave.com/developers/deployed-contracts)
+    ILendingPoolAddressesProvider public constant addressesProvider = ILendingPoolAddressesProvider(0xB53C1a33016B2DC2fF3653530bfF1848a515c8c5);
```
Do we want to initialize these in the constructor, where the constructor accepts the contract addresses? When we deploy, we can specify the contract addresses for either Mainnet or Kovan.

janclarin commented on contracts/FanDonation.sol:
```diff
@@ -0,0 +1,30 @@
+    // Set the address of the AAVE protocol addresses provider. Should not change once deployed.
+    // Currently set to the LendingPoolAddressesProvider for Kovan (https://docs.aave.com/developers/deployed-contracts)
+    ILendingPoolAddressesProvider public constant addressesProvider = ILendingPoolAddressesProvider(0xB53C1a33016B2DC2fF3653530bfF1848a515c8c5);
+    ILendingPool lendingPool = ILendingPool(addressesProvider.getLendingPool());
```
The linter will complain about not specifying the visibility for these state vars, maybe set them to private/internal

janclarin commented on contracts/FanDonation.sol:
```diff
@@ -0,0 +1,30 @@
+    event DonationCompleteStatus(bool success);
+
+    function donate(uint256 amount, address _onBehalfOf) external {
+        IERC20(asset).safeApprove(address(lendingPool), amount);
```
Will this appear as two transactions for the user? One to approve first, then the deposit transaction afterwards?

If not, we may need to separate these into two functions that the front-end will call? approve(asset) and donate(asset, amount, address)? I'm not 100% sure if this works within the same function automatically, but would be great if it does.

Do we want to require() that the fan has approved the ERC20 here, i.e. IERC20(asset).allowance(fan, thisContract) > 0, if we do separate this into two functions?

janclarin replied:
Oops, just realized that safeApprove already checks the last part of my comment, so ignore that part!

janclarin reviewed (changes requested):
Great work! Do you want to wait to submit this until after it's rebased onto the solidity linter PR #11, and optionally the top-level package.json PR #7 to use the openzeppelin npm deps?

janclarin committed:
Add yarn configs

janclarin committed:
Configure yarn workspaces and scripts

janclarin committed:
Replace package-lock.json with yarn.lock

janclarin committed:
Replace package-lock.json with yarn.lock

janclarin committed:
Update CI action to use yarn

janclarin committed:
Update prettierignore to ignore yarn

janclarin committed:
Replace pre- scripts with compile script

Not supported in yarn.

janclarin committed:
Update README for yarn

Removes dependencies section since dependencies are already specified in
package.json

janclarin committed:
Remove npm-run-all commands

janclarin committed:
Merge pull request #8 from janclarin/fan-page

Add fan support card

janclarin committed:
Merge pull request #9 from janclarin/html-title

Update html title and description

chailey committed:
[feature] Add the disconnect wallet feature

chailey committed:
[feature] Add the disconnect wallet feature

chailey committed:
[feature] Hopefully final commit for this branch!

chailey committed:
[feature] Hopefully final commit for this branch!

chailey committed:
Fix merge conflicts

chailey committed:
Oops! No API keys pushed. Created an env file and modified the gitignore subsequently

chailey committed:
Oops! No API keys pushed. Created an env file and modified the gitignore subsequently

janclarin committed:
Run local prettier with yarn for lint/fix

chailey committed:
Fix stuff

chailey committed:
fix stuff

chailey committed:
More fixes

janclarin committed:
Merge pull request #13 from janclarin/use-yarn

Switch to yarn

janclarin committed:
Merge pull request #11 from janclarin/solhint-ci

Add solidity linting

chailey committed:
Merge pull request #15 from janclarin/web3modal

New page + Connect Wallet!

priyankaranke committed:
Merge fix
